cdap-ui/app/hydrator/routes.js

```diff
@@ -125,52 +128,52 @@ angular.module(PKG.name + '.feature.hydrator')
                       if (isVersionInRange) {
                         draft.artifact.version = rCDAPVersion;
                       } else {
-                        defer.resolve(false);
+                        defer.resolve({ valid: false, config: draft, upgrade: true });
```

Comment:
not sure why my comment disappeared on this one..

Why is `valid: false` in here? what is considered `valid`? If you already have a flag for upgrade, can we just simply use that instead of adding another flag?

Reply:
We used to resolve false to indicate that the configuration is invalid/when an error occurs. If we remove this field altogether, we would miss cases where it's not really a pipeline upgrade/import but an error like when parsing JSON.

I can rename it as `validConfig` to make it more readable if you think that helps, please let me know.